### PR TITLE
Update UTM tags in static embed modal links

### LINKED ISSUE #45422: Update all UTM tag to the new version
https://www.notion.so/metabase/Product-UTM-Instrumentation-fc47f3584d5b48bea2e0ad02cda0376d

This issue was raised for [all links other than the upsell banner](https://metaboat.slack.com/archives/C063Q3F1HPF/p1720697648382679), so let's just update those links there to be more specific.

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes https://github.com/metabase/metabase/issues/45422

### Description

Update the upgrade URL for all places that call `getUpgradeUrl` with the following change:
- change `utm_media` to `utm_content`
- change `utm_source` to `source_plan`
- add `utm_source=product`
- add `utm_medium=upsell`
- add `utm_campaign` (only for the link in the static embed flow for simplicity)

Update the UTM tags following our [new format](https://www.notion.so/metabase/Product-UTM-Instrumentation-fc47f3584d5b48bea2e0ad02cda0376d) in the following 4 places in static embed flow:

Screenshot 2024-08-22 at 5 49 46 PM

#### New link
https://www.metabase.com/product/embedded-analytics?utm_source=product&utm_medium=upsell&utm_campaign=embedding-interactive&utm_content=static-embed-popover&source_plan=oss

#### Old link
https://www.metabase.com/product/embedded-analytics?utm_source=oss&utm_media=static-embed-popover

Screenshot 2024-08-22 at 5 51 46 PM

#### New link
https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-overview&source_plan=oss

#### Old link
https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=oss&utm_media=static-embed-settings-overview

Screenshot 2024-08-22 at 5 52 34 PM

#### New link
https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-look-and-feel&source_plan=oss#customizing-the-appearance-of-static-embeds

#### Old link
https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=oss&utm_media=static-embed-settings-appearance#customizing-the-appearance-of-static-embeds

Screenshot 2024-08-22 at 5 52 55 PM

#### New link
https://www.metabase.com/upgrade/?utm_source=product&utm_medium=upsell&utm_campaign=embedding-static-font&utm_content=static-embed-settings-look-and-feel&source_plan=oss&utm_users=7

#### Old link
https://www.metabase.com/upgrade/?utm_media=static-embed-settings-appearance&utm_source=oss&utm_users=7

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by albertoperdomo — APPROVED** [review]

File: e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js [hunk 1]
```diff
@@ -239,7 +239,7 @@ describe("embed modal display", () => {
         cy.findByTestId("interactive-embedding-cta").should(
           "have.attr",
           "href",
-          "https://www.metabase.com/product/embedded-analytics?utm_source=oss&utm_media=static-embed-popover",
+          "https://www.metabase.com/product/embedded-analytics?utm_source=product&utm_medium=upsell&utm_campaign=embedding-interactive&utm_content=static-embed-popover&source_plan=oss",
```
Comment (WiNloSt) [inline e2e/test/scenarios/sharing/public-sharing-embed-button-behavior.cy.spec.js:242]: This is a test for
✅ Interactive embedding upsell
Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ describe("Static Embed Setup phase", () => {
         expect(link).toBeVisible();
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=oss&utm_media=static-embed-settings-overview",
+          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-overview&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:97]: This is a test for
Interactive embedding upsell
✅ Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx [hunk 3]
```diff
@@ -397,7 +397,7 @@ describe("Static Embed Setup phase", () => {
         expect(link).toBeVisible();
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=oss&utm_media=static-embed-settings-appearance#customizing-the-appearance-of-static-embeds",
+          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-look-and-feel&source_plan=oss#customizing-the-appearance-of-static-embeds",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:400]: This is a test for
Interactive embedding upsell
Overview doc
✅ Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx [hunk 4]
```diff
@@ -482,7 +482,7 @@ describe("Static Embed Setup phase", () => {
         expect(link).toBeVisible();
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/upgrade?utm_media=static-embed-settings-appearance&utm_source=oss",
+          "https://www.metabase.com/upgrade?utm_source=product&utm_medium=upsell&utm_campaign=embedding-static-font&utm_content=static-embed-settings-look-and-feel&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/common.unit.spec.tsx:485]: This is a test for
Interactive embedding upsell
Overview doc
Look and Feel doc
✅ Look and Feel font upsell

File: frontend/src/metabase/selectors/settings.ts [hunk 5]
```diff
@@ -97,10 +97,21 @@ export const getUpgradeUrl = createSelector(
   (state: State) => getPlan(getSetting(state, "token-features")),
   (state: State) => getSetting(state, "active-users-count"),
   (state: State, opts: UpgradeUrlOpts) => opts.utm_media,
-  (source, count, media) => {
+  (plan, count, content) => {
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/settings.ts:100]: Just a rename to make them make more sense.

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx [hunk 6]
```diff
@@ -36,7 +36,15 @@ const useCTAText = () => {
     // eslint-disable-next-line no-literal-metabase-strings -- This only shows for admins
     description: t`Give your customers the full power of Metabase in your own app, with SSO, advanced permissions, customization, and more.`,
     linkText: t`Learn more`,
-    url: `https://www.metabase.com/product/embedded-analytics?utm_source=${plan}&utm_media=static-embed-popover`,
+    url: `https://www.metabase.com/product/embedded-analytics?${new URLSearchParams(
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:39]: Every place is using `URLSearchParams` with an object initializer to make UTM code more readable than a long string.

File: enterprise/frontend/src/metabase-enterprise/license/components/LicenseAndBillingSettings/LicenseAndBillingSettings.tsx [hunk 7]
```diff
@@ -173,7 +173,7 @@ const LicenseAndBillingSettings = ({
 export default connect(
   (state: State): StateProps => ({
     settingValues: state.admin.settings.settings,
-    upgradeUrl: getUpgradeUrl(state, { utm_media: "license" }),
+    upgradeUrl: getUpgradeUrl(state, { utm_content: "license" }),
```
Comment (WiNloSt) [inline enterprise/frontend/src/metabase-enterprise/license/components/LicenseAndBillingSettings/LicenseAndBillingSettings.tsx:176]: There is one link in the static embed modal that calls `getUpgradeUrl` so I have to change it everywhere.

File: frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.unit.spec.tsx [hunk 8]
```diff
@@ -36,7 +36,7 @@ describe("DataPermissionsHelp", () => {
     screen.getAllByText("Upgrade to Pro").every(link => {
       expect(link).toHaveAttribute(
         "href",
-        "https://www.metabase.com/upgrade?utm_media=admin_permissions&utm_source=oss",
+        "https://www.metabase.com/upgrade?utm_source=product&utm_medium=upsell&utm_content=admin_permissions&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/permissions/components/DataPermissionsHelp/DataPermissionsHelp.unit.spec.tsx:39]: Upgrade URL change. Please see the issue [description how it has changed](https://github.com/metabase/metabase/pull/47116#issue-2480439695).

File: frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/common.embedding.unit.spec.tsx [hunk 9]
```diff
@@ -39,7 +39,7 @@ describe("[OSS] embedding settings", () => {
           screen.getByRole("link", { name: "upgrade to a paid plan" }),
         ).toHaveProperty(
           "href",
-          "https://www.metabase.com/upgrade?utm_media=embed-settings&utm_source=oss",
+          "https://www.metabase.com/upgrade?utm_source=product&utm_medium=upsell&utm_content=embed-settings&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/common.embedding.unit.spec.tsx:42]: Upgrade URL change. Please see the issue [description how it has changed](https://github.com/metabase/metabase/pull/47116#issue-2480439695).

File: frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/enterprise.embedding.unit.spec.tsx [hunk 10]
```diff
@@ -48,7 +48,7 @@ describe("[EE, no token] embedding settings", () => {
           screen.getByRole("link", { name: "upgrade to a paid plan" }),
         ).toHaveProperty(
           "href",
-          "https://www.metabase.com/upgrade?utm_media=embed-settings&utm_source=oss",
+          "https://www.metabase.com/upgrade?utm_source=product&utm_medium=upsell&utm_content=embed-settings&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/settings/app/components/SettingsEditor/tests/embedding/enterprise.embedding.unit.spec.tsx:51]: Upgrade URL change. Please see the issue [description how it has changed](https://github.com/metabase/metabase/pull/47116#issue-2480439695).

File: frontend/src/metabase/selectors/selectors.unit.spec.ts [hunk 11]
```diff
@@ -12,38 +12,42 @@ import { getIsPaidPlan, getUpgradeUrl } from "./settings";
 describe("getUpgradeUrl", () => {
   it.each([
     {
-      media: "license",
+      content: "license",
       users: 1,
       features: { hosting: false, sso_jwt: false },
-      params: "?utm_media=license&utm_source=oss&utm_users=1",
+      params:
+        "?utm_source=product&utm_medium=upsell&utm_content=license&source_plan=oss&utm_users=1",
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/selectors.unit.spec.ts:19]: Upgrade URL change. Please see the issue [description how it has changed](https://github.com/metabase/metabase/pull/47116#issue-2480439695).

File: frontend/src/metabase/selectors/settings.ts [hunk 12]
```diff
@@ -90,17 +90,29 @@ export const getDocsUrlForVersion = (
...
+  utm_campaign?: string;
+  utm_content: string;
 }
 
 export const getUpgradeUrl = createSelector(
   (state: State) => getPlan(getSetting(state, "token-features")),
   (state: State) => getSetting(state, "active-users-count"),
-  (state: State, opts: UpgradeUrlOpts) => opts.utm_media,
-  (source, count, media) => {
+  (_state: State, utmTags: UpgradeUrlOpts) => utmTags,
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/settings.ts:100]: rename `state` to `_state` to make it clear it's unused.

File: frontend/src/metabase/selectors/settings.ts [hunk 13]
```diff
@@ -90,17 +90,29 @@ export const getDocsUrlForVersion = (
...
+  (plan, count, utmTags) => {
     const url = new URL("https://www.metabase.com/upgrade");
-    url.searchParams.append("utm_media", media);
-    url.searchParams.append("utm_source", source);
+    const searchParams = {
```
Comment (WiNloSt) [inline frontend/src/metabase/selectors/settings.ts:103]: Upgrade URL change. Please see the issue [description how it has changed](https://github.com/metabase/metabase/pull/47116#issue-2480439695).

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx [hunk 14]
```diff
@@ -33,7 +33,7 @@ describe("Static Embed Setup phase - EE, with token", () => {
         expect(link).toBeVisible();
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=pro-self-hosted&utm_media=static-embed-settings-overview",
+          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-overview&source_plan=pro-self-hosted",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx:36]: This is a test for
Interactive embedding upsell
✅ Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx [hunk 15]
```diff
@@ -100,7 +100,7 @@ describe("Static Embed Setup phase - EE, with token", () => {
         expect(link).toBeVisible();
         expect(link).toHaveAttribute(
           "href",
-          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=pro-self-hosted&utm_media=static-embed-settings-appearance#customizing-the-appearance-of-static-embeds",
+          "https://www.metabase.com/docs/latest/embedding/static-embedding.html?utm_source=product&utm_medium=docs&utm_campaign=embedding-static&utm_content=static-embed-settings-look-and-feel&source_plan=pro-self-hosted#customizing-the-appearance-of-static-embeds",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/tests/premium.unit.spec.tsx:103]: This is a test for
Interactive embedding upsell
Overview doc
✅ Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/test/common.unit.spec.tsx [hunk 16]
```diff
@@ -16,7 +16,7 @@ describe("InteractiveEmbeddingCTA", () => {
 
     expect(screen.getByTestId("interactive-embedding-cta")).toHaveAttribute(
       "href",
-      "https://www.metabase.com/product/embedded-analytics?utm_source=oss&utm_media=static-embed-popover",
+      "https://www.metabase.com/product/embedded-analytics?utm_source=product&utm_medium=upsell&utm_campaign=embedding-interactive&utm_content=static-embed-popover&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/test/common.unit.spec.tsx:19]: This is a test for
✅ Interactive embedding upsell
Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/test/enterprise.unit.spec.tsx [hunk 17]
```diff
@@ -25,7 +25,7 @@ describe("InteractiveEmbeddingCTA", () => {
 
     expect(screen.getByTestId("interactive-embedding-cta")).toHaveAttribute(
       "href",
-      "https://www.metabase.com/product/embedded-analytics?utm_source=oss&utm_media=static-embed-popover",
+      "https://www.metabase.com/product/embedded-analytics?utm_source=product&utm_medium=upsell&utm_campaign=embedding-interactive&utm_content=static-embed-popover&source_plan=oss",
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/test/enterprise.unit.spec.tsx:28]: This is a test for
✅ Interactive embedding upsell
Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/OverviewSettings.tsx [hunk 18]
```diff
@@ -76,7 +76,13 @@ export const OverviewSettings = ({
           <Text>{jt`Check out the ${(
             <ExternalLink
               key="doc"
-              href={`${docsUrl}?utm_source=${plan}&utm_media=static-embed-settings-overview`}
+              href={`${docsUrl}?${new URLSearchParams({
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/OverviewSettings.tsx:79]: This is a change for
Interactive embedding upsell
✅ Overview doc
Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 19]
```diff
@@ -72,7 +74,13 @@ export const LookAndFeelSettings = ({
           <Text>{jt`These options require changing the server code. You can play around with and preview the options here. Check out the ${(
             <ExternalLink
               key="doc"
-              href={`${docsUrl}${utmTags}#customizing-the-appearance-of-static-embeds`}
+              href={`${docsUrl}?${new URLSearchParams({
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:77]: This is a change for
Interactive embedding upsell
Overview doc
✅ Look and Feel doc
Look and Feel font upsell

File: frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx [hunk 20]
```diff
@@ -52,7 +52,10 @@ export const LookAndFeelSettings = ({
     }),
   );
   const upgradePageUrl = useSelector(state =>
-    getUpgradeUrl(state, { utm_media: "static-embed-settings-appearance" }),
+    getUpgradeUrl(state, {
```
Comment (WiNloSt) [inline frontend/src/metabase/public/components/EmbedModal/StaticEmbedSetupPane/LookAndFeelSettings.tsx:55]: This is a change for
Interactive embedding upsell
Overview doc
Look and Feel doc
✅ Look and Feel font upsell

**Review by albertoperdomo — APPROVED** [review]

**Comment by WiNloSt** [comment]
We [decided](https://metaboat.slack.com/archives/C063Q3F1HPF/p1724399768211849?thread_ts=1724077872.663639&cid=C063Q3F1HPF) to backport this to have cleaner data sooner than later

File: frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx (hunk 6 above)
Comment (heypoom) [reply frontend/src/metabase/public/components/EmbedModal/SelectEmbedTypePane/InteractiveEmbeddingCTA/InteractiveEmbeddingCTA.tsx:39]: (TIL that `URLSearchParams` implicitly casts to string params by default)

**Review by heypoom — APPROVED** [review]
Reviewed all the changes - looks correct to me!

**Pull request merged**